peer/sync.go

```diff
@@ -82,6 +82,13 @@ func (syncer *Syncer) Sync(ctx context.Context, contentID []byte, hint *id.Signa
 	}
 	syncer.pendingMu.Unlock()
 
+	// Ensure that pending content is removed.
+	defer func() {
+		syncer.pendingMu.Lock()
+		delete(syncer.pending, string(contentID))
+		syncer.pendingMu.Unlock()
+	}()
```

Comment:
You need to add a test that fails if this code is removed, but passes if this code is present. Otherwise, we cannot be sure that this has actually fixed the problem it is supposed to fix.

Reply:
Not sure how I would test this since the map itself is a private field

Reply:
@rahulghangas You don't need to interact with the field. Just need a general syncing test (like the example I posted in Slack) that fails if that code is commented out, and passes when the code is added.

Reply:
The purpose of this block is to make sure that a pending struct is deleted once a sync message is successfully received or a sync request fails. To make sure that it is indeed deleted, I need to check the map for the corresponding `contentID` once the function is exited

Reply:
@loongy not sure how to test this

Reply:
@rahulghangas You could have a general test that fails without it and succeeds once you have it. For example:

1. N nodes in the network
2. Node A has some data
3. Node B attempts to sync this data
4. Node B’s context expires so it is unable to sync the data
5. Node B attempts to sync again, but it realises there is already a pending channel it can wait on
6. Node B decides to wait on this channel, and never receives the content because the original sync attempt has already expired

This test would fail without the pending map clean-up, but should pass once you add it.